pkg/app/master/command/cliparamsfile.go

```diff
@@ -0,0 +1,148 @@
+	"errors"
+	"fmt"
+	"os"
+	"reflect"
+	"strconv"
+	"strings"
+
+	"github.com/mintoolkit/mint/pkg/app"
+	"github.com/urfave/cli/v2"
+)
+
+func ParseParamsFromFile(ctx *cli.Context, xc *app.ExecutionContext, filePath string, validFlags []cli.Flag) {
+	params, err := parseJsonFileParams(filePath)
+
+	if err != nil {
+		xc.Out.Error("params.file.error", err.Error())
+		xc.Out.State("exited",
+			ovars{
+				"exit.code": -1,
+			})
+		xc.Exit(-1)
+	}
+
+	err = setFileParams(ctx, xc, params, validFlags)
+
+	if err != nil {
+		xc.Out.Error("params.file.param.value.error", err.Error())
+		xc.Out.State("exited",
+			ovars{
+				"exit.code": -1,
+			})
+		xc.Exit(-1)
+	}
+
+}
+
+func validFlagsMap(validFlags []cli.Flag) (map[string]cli.Flag, error) {
+	validFlagsMap := make(map[string]cli.Flag)
+
+	for _, flag := range validFlags {
+		if len(flag.Names()) == 0 {
+			return nil, errors.New("flag names has no values")
+		}
+
+		validFlagsMap[flag.Names()[0]] = flag
+	}
+
+	return validFlagsMap, nil
+}
+
+func flagSetString(flag cli.Flag, paramValue interface{}) (string, error) {
+	paramValueType := reflect.TypeOf(paramValue).Kind()
+
+	switch flag.(type) {
+	case *cli.StringFlag:
+		if paramValueType == reflect.String {
+			return paramValue.(string), nil
+		} else {
+			return "", fmt.Errorf("expected string value found: %v for flag %s", paramValueType, flag.Names()[0])
+		}
+	case *cli.BoolFlag:
+		if paramValueType == reflect.Bool {
+			return strconv.FormatBool(paramValue.(bool)), nil
+		} else {
+			return "", fmt.Errorf("expected boolean value found: %v for flag %s", paramValueType, flag.Names()[0])
+		}
+	case *cli.IntFlag:
+		// JSON Unmarshal unmarshals a JSON number as a Float64 by default
+		if paramValueType == reflect.Int {
+			return strconv.Itoa(paramValue.(int)), nil
+		} else if paramValueType == reflect.Float64 {
+			// Decimal will be truncated out here
+			return strconv.Itoa(int(paramValue.(float64))), nil
+		} else {
+			return "", fmt.Errorf("expected Int value found: %v for flag %s", paramValueType, flag.Names()[0])
+		}
+	case *cli.StringSliceFlag:
+		// Parse both string slice or string values
+		if paramValueType == reflect.String {
+			return paramValue.(string), nil
+		} else if paramValueType == reflect.Slice {
+			var values []string
+			slice := reflect.ValueOf(paramValue)
+			for i := 0; i < slice.Len(); i++ {
+				values = append(values, fmt.Sprintf("%v", slice.Index(i).Interface()))
+			}
+			return strings.Join(values, ","), nil
+		} else {
+			return "", fmt.Errorf("expected string or string slice value found: %v for flag %s", paramValueType, flag.Names()[0])
+		}
+	default:
+		return "", errors.New("flag type not supported by params file")
+	}
+}
+
+func setFileParams(ctx *cli.Context, xc *app.ExecutionContext, params map[string]interface{}, validFlags []cli.Flag) error {
+	validFlagsMap, err := validFlagsMap(validFlags)
+
+	if err != nil {
+		return err
+	}
+
+	for fileParamKey, fileParamValue := range params {
```

Comment:
Flag names as JSON file field names don't look very natural. Doing an explicit map for each flag is a lot of work and it's less than ideal at this point, so the next best thing is to do a generic transform on the name. JSON field names especially in JSON/REST API calls normally use underscores in field names (at least, the ones implemented in python, ruby and often enough in node). Let's go with that naming convention, so for a flag like "include-path" the corresponding JSON file field name would be "include_path" and "include-app-image-all" would be "include_app_image_all".

The reason I mentioned JSON/REST APIs is because that's one of the follow up enhancements to have the server mode where each command will be a REST/JSON API endpoint and the HTTP body request parameters in those calls would be an equivalent of providing the `--command-params-file` flag.

Reply:
Agree JSON files using the kebab case just look odd, I have commited a function to handle the transform from snake case to kebab case (still works with kebab case if a file using kebab case commands is input). 

Good to know about the future enhancement - hopefully some of these functions can be reused for its implementation.